src/ncp/ncp_base.cpp

```diff
@@ -3161,6 +3164,64 @@ ThreadError NcpBase::SetPropertyHandler_THREAD_CONTEXT_REUSE_DELAY(uint8_t heade
     return errorCode;
 }
 
+#if OPENTHREAD_ENABLE_DIAG
+ThreadError NcpBase::SetPropertyHandler_NEST_STREAM_MFG(uint8_t header, spinel_prop_key_t key, const uint8_t *value_ptr, uint16_t value_len)
+{
+    char *string(NULL);
+    char *output(NULL);
+    spinel_ssize_t parsedLength;
+    ThreadError errorCode = kThreadError_None;
+
+    parsedLength = spinel_datatype_unpack(
+                       value_ptr,
+                       value_len,
+                       SPINEL_DATATYPE_UTF8_S,
+                       &string
+                   );
+
+    if ((parsedLength > 0) && (string != NULL))
+    {
+        char *argv[8];
+        int argc = 0;
+        int length = static_cast<int>(strlen(string));
+        char *cmd;
+
+        for (; *string == ' '; string++, length--);
+
+        argv[argc++] = string;
+
+        for (cmd = string + 1; (cmd < string + length) && (cmd != NULL); ++cmd)
+        {
+            if (*cmd == ' ' || *cmd == '\r' || *cmd == '\n')
+            {
+                *cmd = '\0';
+            }
+
+            if (*(cmd - 1) == '\0' && *cmd != ' ')
+            {
+                argv[argc++] = cmd;
+            }
+        }
+
+        // all diagnostics related features are processed within diagnostics module
+        output = diagProcessCmd(argc, argv);
+
+        errorCode = SendPropertyUpdate(
+                        header,
+                        SPINEL_CMD_PROP_VALUE_IS,
+                        key,
+                        SPINEL_DATATYPE_UTF8_S,
+                        output
```

Comment:
`SendPropertyUpdate()` is overloaded. One of the overloaded versions takes just a data pointer and a data length. In this case, since the length of the string could be rather long, it would be more appropriate to use that version of the function:

```
errorCode = SendPropertyUpdate(
    header,
    SPINEL_CMD_PROP_VALUE_IS,
    key,
    output,
    strlen(output) + 1 // +1 to include the zero-termination
);
```